src/Host/Client/Impl/Host/LocalRHostConnector.cs

```diff
@@ -31,134 +33,150 @@ internal sealed class LocalRHostConnector : IRHostConnector {
-                let pos = rnd.NextDouble()
-                orderby pos
-                select port;
-
-            var transportTcs = new TaskCompletionSource<IMessageTransport>();
-            foreach (var port in ports) {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                server = new WebSocketServer(port) {
-                    ReuseAddress = false,
-                    WaitTime = HeartbeatTimeout,
-                };
-
-                server.AddWebSocketService("/", () => CreateWebSocketMessageTransport(transportTcs));
-
-                try {
-                    server.Start();
-                    break;
-                } catch (SocketException ex) {
-                    if (ex.SocketErrorCode == SocketError.AddressAlreadyInUse) {
-                        server = null;
-                    } else {
-                        throw new MessageTransportException(ex);
+            base.Dispose();
+
+            if (_brokerProcess != null) {
+                if (!_brokerProcess.HasExited) {
+                    try {
+                        _brokerProcess.Kill();
+                    } catch (Win32Exception) {
+                    } catch (InvalidOperationException) {
                     }
-                } catch (WebSocketException ex) {
-                    throw new MessageTransportException(ex);
-                }
-            }
 
-            if (server == null) {
-                throw new MessageTransportException(new SocketException((int)SocketError.AddressAlreadyInUse));
+                    _brokerProcess = null;
+                }
             }
+        }
 
-            var psi = new ProcessStartInfo {
-                FileName = rhostExe,
-                UseShellExecute = false
+        protected override HttpClientHandler GetHttpClientHandler() {
+            return new HttpClientHandler {
+                Credentials = _credentials
             };
+        }
 
-            var shortHome = new StringBuilder(NativeMethods.MAX_PATH);
-            NativeMethods.GetShortPathName(_rHome, shortHome, shortHome.Capacity);
-            psi.EnvironmentVariables["R_HOME"] = shortHome.ToString();
+        protected override void ConfigureWebSocketRequest(HttpWebRequest request) {
+            request.AuthenticationLevel = AuthenticationLevel.MutualAuthRequested;
+            request.Credentials = _credentials;
+        }
 
-            psi.EnvironmentVariables["PATH"] = rBinPath + ";" + Environment.GetEnvironmentVariable("PATH");
+        protected override async Task ConnectToBrokerAsync() {
+            DisposableBag.ThrowIfDisposed();
+            await TaskUtilities.SwitchToBackgroundThread();
 
-            if (name != null) {
-                psi.Arguments += " --rhost-name " + name;
+            try {
+                if (await _connectLock.WaitAsync()) {
+                    if (!_isConnected) {
```

Comment:
Let's see how frequently this is an issue first. One attempt approach solves the "I killed broker from Task Manager" problem, but will make repeating broker crashes more visible, which is probably what we want at least while we're testing it. 

---
In reply to: [75047226](https://github.com/Microsoft/RTVS/pull/2173#discussion_r75047226) [](ancestors = 75047226,75034380,75033980,75033083,75031624)